assets/js/components/Dashboard/PlaceRow.tsx

```diff
@@ -89,7 +89,20 @@ const PlaceRow = (props: PlaceRowProps): JSX.Element => {
       .map((screen) => [screen]);
 
     if (paEssScreens.length > 0) {
-      groupedScreens.push(paEssScreens);
+      const paEssGroupedByRoute = new Map<string, Screen[]>();
+      paEssScreens.map((paEssScreen) => {
+        if (paEssScreen.station_code) {
+          const routeLetter = paEssScreen.station_code.charAt(0);
+
+          paEssGroupedByRoute.has(routeLetter)
+            ? paEssGroupedByRoute.get(routeLetter)?.push(paEssScreen)
+            : paEssGroupedByRoute.set(routeLetter, [paEssScreen]);
+        }
+      });
+
+      paEssGroupedByRoute.forEach((screens) => {
+        groupedScreens.push(screens);
+      });
```

Comment:
To help simplify the logic that lives in this function, I would pull this logic out into it's own helper function.